src/util-ebpf.c

```diff
@@ -533,56 +535,97 @@ static bool EBPFCreateFlowForKey(struct flows_stats *flowstats, FlowKey *flow_ke
-    if (f != NULL) {
-        SCLogDebug("bypassed flow found %d -> %d, doing accounting",
-                    f->sp, f->dp);
-        FlowCounters *fc = FlowGetStorageById(f, GetFlowBypassCounterID());
-        if (fc == NULL) {
-            FLOWLOCK_UNLOCK(f);
-            flowstats->count++;
+    int i;
+    uint64_t pkts_cnt = 0;
+    uint64_t bytes_cnt = 0;
+    /* We use a per CPU structure so we will get a array of values. But if nr_cpus
+     * is 1 then we have a global hash. */
+    BPF_DECLARE_PERCPU(struct pair, values_array, eb->cpus_count);
+    memset(values_array, 0, sizeof(values_array));
+    int res = bpf_map_lookup_elem(eb->mapfd, key, values_array);
+    if (res < 0) {
+        SCLogDebug("errno: (%d) %s", errno, strerror(errno));
+        return false;
+    }
+    for (i = 0; i < eb->cpus_count; i++) {
+        /* let's start accumulating value so we can compute the counters */
+        SCLogDebug("%d: Adding pkts %lu bytes %lu", i,
+                BPF_PERCPU(values_array, i).packets,
+                BPF_PERCPU(values_array, i).bytes);
+        pkts_cnt += BPF_PERCPU(values_array, i).packets;
+        bytes_cnt += BPF_PERCPU(values_array, i).bytes;
+    }
+    FlowBypassInfo *fc = FlowGetStorageById(f, GetFlowBypassInfoID());
+    if (fc == NULL) {
+        return false;
+    }
+    if (index == 0) {
+        if (pkts_cnt != fc->todstpktcnt) {
+            fc->todstpktcnt = pkts_cnt;
+            fc->todstbytecnt = bytes_cnt;
             return true;
         }
-        if (flow_key->sp == f->sp) {
-            if (pkts_cnt != fc->todstpktcnt) {
-                flowstats->packets += pkts_cnt - fc->todstpktcnt;
-                flowstats->bytes += bytes_cnt - fc->todstbytecnt;
-                fc->todstpktcnt = pkts_cnt;
-                fc->todstbytecnt = bytes_cnt;
-                /* interval based so no meaning to update the millisecond.
-                 * Let's keep it fast and simple */
-                f->lastts.tv_sec = ctime->tv_sec;
-            }
-        } else {
-            if (pkts_cnt != fc->tosrcpktcnt) {
-                flowstats->packets += pkts_cnt - fc->tosrcpktcnt;
-                flowstats->bytes += bytes_cnt - fc->tosrcbytecnt;
-                fc->tosrcpktcnt = pkts_cnt;
-                fc->tosrcbytecnt = bytes_cnt;
-                f->lastts.tv_sec = ctime->tv_sec;
-            }
+    } else {
+        if (pkts_cnt != fc->tosrcpktcnt) {
+            fc->tosrcpktcnt = pkts_cnt;
+            fc->tosrcbytecnt = bytes_cnt;
+            return true;
         }
-        FLOWLOCK_UNLOCK(f);
+    }
+
+    return false;
+}
+
+/** Check both half flows for update
+ *
+ * Update lastts in the flow and do accounting
+ *
+ * */
+bool EBPFBypassUpdate(Flow *f, void *data, time_t tsec)
+{
+    EBPFBypassData *eb = (EBPFBypassData *)data;
+    if (eb == NULL) {
         return false;
+    }
+    bool activity = EBPFBypassCheckHalfFlow(f, eb, eb->key[0], 0);
```

Comment:
How about something like:
```c
/* call both unconditionally to update counters */
bool activity = EBPFBypassCheckHalfFlow(f, eb, eb->key[0], 0);
activity |= EBPFBypassCheckHalfFlow(f, eb, eb->key[1], 1));
```

Reply:
Are we sure an optimization will not skip the eval if activity is already true ?

Reply:
I don't think it should. We use this pattern elsewhere too.